packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -186,7 +199,36 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
     /**
      * {@inheritDoc (ISharedObject:interface).snapshot}
      */
-    public abstract snapshot(): ITree;
+    public summarize(fullTree: boolean = false): IChannelSummarizeResult {
+        // Create a SummarySerializer that will be used to serialize any IFluidHandles in this object. The
+        // IFluidHandles represents route to any referenced Fluid object.
+        // SummarySerializer tracks the routes of all handles that it serializes.
+        const serializer = new SummarySerializer(this.runtime.channelsRoutingContext);
+
+        const snapshot: ITree = this.snapshotCore(serializer);
+        const summaryTree = convertToSummaryTreeWithStats(snapshot, fullTree);
+
+        const node: IGarbageCollectionNode = {
```

Comment:
I think API should be more flexible and support future changes.
I.e. I think it should be [[node, routes]], i.e. an array of nodes.
This will allow channels to have data stores and have any nesting depth in the future.
DDS can always return just one entry as you do here.

Unrelated (did not get there yet, but sharing here):
Even Data store likely should be able to return just one entry, i.e. essentially aggregate all DDSs into one and not care about sub-nodes splitting. Supporting that would require more logic from GC core, as it needs to understand that if there is route /id1/id2, and there is no such node, then it means it need to "downgrade" that route to be /id1.
This is optional / future issue to implement (if we did not implement it yet).

Reply:
Ah, I see you return an array below :)

Reply:
I removed the code that generates this GC data and sends it all the way to container runtime. That was more of a prototype and I will do that in a separate PR once I have a more definitive format of the gc data that is to be returned here.